utility/include/isobus/utility/thread_synchronization.hpp

```diff
@@ -38,6 +106,95 @@ namespace isobus
 /// @param x The mutex to lock.
 #define LOCK_GUARD(type, x) const std::lock_guard<type> x##Lock(x)
 
+/// @brief A template class for a lock free queue.
+/// @tparam T The item type for the queue.
+template<typename T>
+class LockFreeQueue
+{
+public:
+	/// @brief Constructor for the lock free queue.
+	explicit LockFreeQueue(std::size_t size) :
+	  buffer(size), capacity(size) {}
```

Comment:
Might sound a little insane, but we might want to zero-check the size... I see the possibility of `% 0` later and my UB alarms start going off